mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -40,6 +40,44 @@ static llvm::Value *createDeviceFunctionCall(llvm::IRBuilderBase &builder,
   return builder.CreateCall(fn, args);
 }
 
+static llvm::Value *createSubGroupShuffle(llvm::IRBuilderBase &builder,
+                                          llvm::Value *value, llvm::Value *mask,
+                                          GENX::ShflKind kind) {
+  std::string fnName = "";
```

Comment:
assert that mask as type that is a i32. 
assert that value has type that is an integer or floating point type ?

Reply:
no need tablegen should have already did that for you.

Reply:
Yes I know it did it for the operator but this is a function that might get called somewhere else.

Reply:
added assert for mask's type. Already have unreachable for value's type.